chat-android/src/main/java/com/ably/chat/Typing.kt

```diff
@@ -163,82 +187,132 @@ internal class DefaultTyping(
     override suspend fun get(): Set<String> {
         logger.trace("DefaultTyping.get()")
         room.ensureAttached(logger) // CHA-T2d, CHA-T2c, CHA-T2g
-        return channelWrapper.presence.getCoroutine().map { it.clientId }.toSet()
+        return currentlyTypingMembers
     }
 
     override suspend fun start() {
         logger.trace("DefaultTyping.start()")
-
-        typingScope.launch {
-            // If the user is already typing, reset the timer
-            if (typingJob != null) {
-                logger.debug("DefaultTyping.start(); already typing, resetting timer")
-                typingJob?.cancel()
-                startTypingTimer()
-            } else {
-                startTypingTimer()
-                room.ensureAttached(logger) // CHA-T4a1, CHA-T4a3, CHA-T4a4
-                channelWrapper.presence.enterClientCoroutine(room.clientId)
+        typingScope.async {
+            room.ensureAttached(logger) // CHA-T4a1, CHA-T4a3, CHA-T4a4
+            // If the user is already typing, return
+            if (typingInProgressJob?.isActive == true) {
+                logger.warn("DefaultTyping.start(); already typing, so it's a no-op")
+                return@async
+            }
+            startHeartbeatTimer()
+            try {
+                sendTyping(TypingEventType.Started, room.clientId)
+            } catch (e: Exception) {
+                typingInProgressJob?.cancel()
+                throw e
             }
-        }.join()
+        }.await()
     }
 
     override suspend fun stop() {
         logger.trace("DefaultTyping.stop()")
-        typingScope.launch {
-            typingJob?.cancel()
-            typingJob = null
+        typingScope.async {
```

Comment:
if there is no result, it's better to use `launch`

Reply:
Actually, `stop` needs to throw error. Similar to https://github.com/ably/ably-chat-kotlin/pull/117/files/c5ed4b4212de2178a922a3ecab3311e4fe73ce8a#r2007531199, will double check 👍

Reply:
Same as https://github.com/ably/ably-chat-kotlin/pull/117#discussion_r2009448093